src/config/transports.js

```diff
@@ -18,6 +23,33 @@ class S3FileTransport extends TransportStream {
   }
 
   uploadToS3() {
+    const parsedUrl = new URL(this.signedUrl);
+    const params = new URLSearchParams(parsedUrl.search);
+
+    const amzDate = params.get('X-Amz-Date');
+    const amzExpires = params.get('X-Amz-Expires');
+
+    const dateFormart = 'YYYYMMDDTHHmmss[Z]';
+    const dateExpires = moment.utc(amzDate, dateFormart).toDate();
+    dateExpires.setSeconds(dateExpires.getSeconds(), amzExpires * 1000);
+
+    const now = new Date();
+    if (dateExpires < now) {
+      const requestGetUploadInfo = async () => {
+        const response = await api.getUploadInfo(this.jobInfo.projectId, this.apiKey);
+        if (!response || !response.body) {
+          return null;
+        }
+        const { body } = response;
+        const { uploadUrl } = body;
+        this.signedUrl = uploadUrl;
+        const batch = generateUuid();
+        const fileName = path.basename(this.filePath);
+        return api.saveJobLog(this.jobInfo, batch, fileName, this.apiKey);
+      };
+      requestGetUploadInfo();
```

Comment:
Do we need 'await' here?

Reply:
Added await for  `requestGetUploadInfo();` Thank you.